### PR TITLE
test(compaction_limit): added new feature test

### PR DESCRIPTION
**Pull request by soyacz**
Test measuring impact on ops/latency when compaction limit is set to
different values. In case of ops/latency drop in relation to base
results (no limits applied) raises `ErrorEvent`.

refs: https://github.com/scylladb/qa-tasks/issues/256

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by soyacz** [comment]
test job: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/compaction-throughput-limit-test/9/

**Comment by soyacz** [comment]
Test shows that enforcing limit increases latency and reduces op/s - maybe disk load is too heavy for these instances and we don't comply to recommendations of limit to be x16 higher than throughput. I'll retest with 2xlarge machines to see how it behaves. 
Results can be checked with:
`hydra investigate show-monitor e251a9d4-89fe-47fd-9a4b-960348239e49`
Example graph (can see 3 runs: no limits, 85% limit, 65% limit)
image

**Comment by soyacz** [comment]
I tested with i3.xlarge machines - there's visible improvement when limiting compaction throughput to 60%. Still looks disk is heavily loaded - I'll retry with smaller disk footprint.

image
image

File: jenkins-pipelines/features-per-partition-limit.jenkinsfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
 longevityPipeline(
```
Comment (roydahan) [inline jenkins-pipelines/features-per-partition-limit.jenkinsfile:6]: I think this file isn't related to this PR, is it?

Comment (roydahan) [reply jenkins-pipelines/features-per-partition-limit.jenkinsfile:6]: I see now that it covers two features (not only the one in the title)

Comment (soyacz) [reply jenkins-pipelines/features-per-partition-limit.jenkinsfile:6]: I reused some code from per-partition limit test - so I moved it to different file&class.

**Comment by soyacz** [comment]
I think there's nothing more to do regarding this test. Feature looks working correctly. Of course it must be used with caution.
Graphs from recent test for limits: 
no limit, 283MB/s, 182MB/s, 81MB/s, 4MB/s:
throughput_latency
compaction_throughput

@fruch @roydahan please review

**Comment by fruch** [comment]
> I think there's nothing more to do regarding this test. Feature looks working correctly. Of course it must be used with caution. Graphs from recent test for limits: no limit, 283MB/s, 182MB/s, 81MB/s, 4MB/s: throughput_latency compaction_throughput
> 
> @fruch @roydahan please review

I would argue that 70% and 45% are less interesting, and would like to see more between 20% and %1, where we actually see it in action, and can deduce something about what's the optimal limit for this use-case (i.e. specific cassnadra-stress command)

**Review by fruch — APPROVED** [review]
LGTM

**Review by roydahan — APPROVED** [review]

**Pull request merged**